roles/pulp/README.md

```diff
@@ -96,6 +101,80 @@ Operating System Variables:
 Each currently supported operating system has a matching file in the "vars"
 directory.
 
+Idempotency:
+------------
+This role is idempotent by default. It is dependent on these settings remaining `false`:
+* `pulp_update`
+* Every `update` under `pulp_install_plugins`
+
+Note on Plugin Version Compatibility with Pulpcore
+--------------------------------------------------
+
+Pulp 3 has a plugin architecture so that new content types, and new features, can be added by the larger community. However, both pulpcore & plugins are installed via pip, which has limited dependency resolution. Plugins release at their own lifecycles. Thus in the worst case scenario, the 2 stable branches of plugin pulp_juicy could depend on the current branch of pulpcore, while the 2 stable branches of pulp_sugary could depend on the current branch of pulpcore.
+
+In order to avoid breaking multiple plugins for the sake of 1 plugin, and to avoid breaking existing installs, updating a plugin will not cause pulpcore to be updated as dependency. Similarly, if a plugin were to be attempted to be updated to an incompatible version with pulpcore, the installer will fail & exit.
+
+Thus you, yourself, must research plugin compatibility with pulpcore whether you are installing 1 plugin, or more than 1 plugin.
+
+Suggested Workflows for Pulpcore & Plugin Versioning:
+-----------------------------------------------------
+
+### Latest versions with minimal work:
+
+Initial install:
+
+1. Observe the latest branch of pulpcore (3.0).
+2. Confirm that all the latest stable releases of your desired plugins are compatibile with pulpcore 3.0, such as by reading their README.md.
+3. Do not set `pulp_branch`. Do not set `branch` under `pulp_install_plugins`.
+4. Run ansible-pulp.
+
+Updating your install:
+
+1. Check if `pulp_branch` has changed to a new stable branch in the latest version of the installer. This will happen whenever a new branch of `pulpcore` is released. Let's assume 3.0 is stil the latest, but there are new micro updates (like pulpcore 3.0.3 -> 3.0.4, and pulp_juicy 1.0.3 -> 1.0.4)
```

Comment:
```suggestion
1. Check if `pulp_branch` has changed to a new stable branch in the latest version of the installer. This will happen whenever a new branch of `pulpcore` is released. Let's assume 3.0 is still the latest, but there are new micro updates (like pulpcore 3.0.3 -> 3.0.4, and pulp_juicy 1.0.3 -> 1.0.4)
```